.github/workflows/setup.yml

```
@@ -0,0 +1,59 @@
+name: Project setup
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Set-up PostgreSQL
+        uses: ikalnytskyi/action-setup-postgres@v5
+        with: 
+          postgresql-version: 14
+          port: 5432
+          database: evershop
+          username: postgres
+          password: pgAdmin12345
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Setup database schema
+        run:  |
+          npm run build
+          npm run dev &
+        env:
+          NODE_ENV: production
+          DB_HOST: localhost
+          DB_PORT: 5432
+          DB_NAME: evershop
+          DB_USER: postgres
+          DB_PASSWORD: pgAdmin12345
+          ADMIN_EMAIL: example@gmail.com
+          ADMIN_PASSWORD: pSynidexxx.899
+          SUPPRESS_NO_CONFIG_WARNING: "true"
```

```suggestion
      - name: Setup database schema
        env:
          NODE_ENV: production
          DB_HOST: localhost
          DB_PORT: 5432
          DB_NAME: evershop
          DB_USER: postgres
          DB_PASSWORD: pgAdmin12345
          ADMIN_EMAIL: example@gmail.com
          ADMIN_PASSWORD: pSynidexxx.899
          SUPPRESS_NO_CONFIG_WARNING: "true"
          run:  |
          npm run build
          npm run dev &
```

It would be good to place the env section at the beginning for clarity and consistency. 
This ensures that environment variables are set before any commands are executed.

---

Okay. Thank you for the comment.